core/src/test/java/org/apache/iceberg/encryption/TestKeyMetadataParser.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.encryption;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestKeyMetadataParser {
+
```

Comment:
I think we should add a test for invalid schema (unsupported schema)

Reply:
how would we generate a key metadata buffer with invalid schema? The schema is hard-coded (V1) in the generator class, we will need to duplicate this class for the test.

Reply:
@ggershinsky you can just use a buffer with 0x02 as the first byte.

Reply:
Ok, this will test an unsupported version in the header.